examples/federated-learning/players.py

```diff
@@ -139,6 +140,10 @@ def _update_one_round(self, data_owners, **kwargs):
 
     # Update step
     with self.device:
+      if isinstance(aggr_gradients[0], PondPrivateTensor):
```

Comment:
I'd like to find a better way of addressing thus, eg not using reveal directly

Reply:
Sounds good. Suggestions welcome. Before we were using local computation for all of these functions but we decided to move away from that.

Reply:
this could also be grouped into `_call_aggregator_fn` to be applied automatically after the aggregation, but agree that it's currently too easy for the user to break security by not aggregating sufficiently in their custom `aggregator_fn` (assuming that's why @mortendahl isn't a fan, maybe I'm wrong)

Reply:
> Before we were using local computation for all of these functions but we decided to move away from that.

what was the motivation behind this?

Reply:
> it's currently too easy for the user to break security

security is already broken since the script running currently gets to see the gradients instead of just the model owner

Reply:
I see it as a lower-lever primitive that makes it too easy to make mistakes: if you're not calling it while pinned to the right device you may end up leaking sensitive information to eg the script running (initially thought that happened here, until I saw the `pin_to_owner` specified several levels away).

Reply:
So what's the solution here? That the original was secure but not the ideal way of doing it? Should we just be using local_computation?

Reply:
yeah I think it's secure (should be verified via the graph)!